degica/omni_exchange
# Return plain format fx rate

## Context


Addresses [this issue in #14823](https://github.com/degica/hats/pull/14823#discussion_r955555300)

Some currencies, like USD and EUR, have subunits (ie. they use cents). So, when doing foreign exchange conversion, $1 should be written out as 100 USD, not 1 USD.

In order to provide an accurate foreign exchange rate for currencies with subunits, this gem moves the decimal point of an exchange rate depending on the base currency's subunit. However, there are situations when the user may not want an exchange rate with the decimal point of the exchange rate moved. For example, on show pages where 100 USD is displayed as $1.00, you'd want display an exchange rate that doesn't have the decimal place moved since the currency is being displayed as a decimal.

This PR allows this gem to return an exchange rate that doesn't have its decimal place moved.


## How to test

1. In your terminal, run `bin/console`
2. This repo is public, so I won't be providing any secret credentials in these instructions... sorry! You'll have to go to 1Password, look for Openexchangerates, visit this [link](https://openexchangerates.org/account/app-ids), sign in, and get the `app_id` which will be used in step 3.
3. Configure the gem by running the following. Don't forget to add the Open Exchange Rates `app_id` that you just got in step 2!
`OmniExchange.configure do |config|
  config.provider_config = {
    open_exchange_rates: {
      app_id: "THE APP ID GOES HERE"
    }
  }
end`
4. Use the gem to convert 100 USD ($1) to JPY - `usd_to_jpy = OmniExchange.get_fx_data(amount: 100, base_currency: "USD", target_currency: "JPY", providers: [:open_exchange_rates])`
5. Take note of the converted amount - `test1 = usd_to_jpy[:converted_amount].to_f`
6. Take note of the converted amount when converting $1 using 100 and the old version of the exchange rate - `test2 = 100 * (usd_to_jpy[:exchange_rate].to_f)`
7. Take note of the converted amount when converting $1 using 1.00 and the new version of the exchange rate - `test3 = 1.00 * (usd_to_jpy[:non_subunit_fx_rate].to_f)`
8.  Please check to make sure that all three converted amounts are the same.
<img width="687" alt="image" src="https://user-images.githubusercontent.com/62090256/187588978-88e22b50-3d0b-4237-bf6e-9b3508e8c1f5.png">
